EIPS/eip-3450.md

```diff
@@ -0,0 +1,151 @@
+that satisfies these conditions is uniquely determined and can be obtained using the Lagrange interpolation formula given below.
+
+Shamir's secret sharing scheme is applied separately to each of the _n_ words (or, the integer representation of each word) of the shared mnemonic, so we work with
+_y<sub>i</sub>_ as a vector of _n_ values, where _y<sub>i</sub>_[_l_] = _f<sub>l</sub>_(_x<sub>i</sub>_), 1 ≤ _l_ ≤ _n_,
+and _f<sub>j</sub>_ is the polynomial in the _j_-th instance of the scheme.
+
+Interpolate(_x_, {(_x<sub>i</sub>_, _y<sub>i</sub>_), 1 ≤ _i_ ≤ _K_})
+
+Input: the desired index _x_, a set of index/value-vector pairs {(_x<sub>i</sub>_, _y<sub>i</sub>_), 1 ≤ _i_ ≤ _K_} ⊆ GF(2048) × GF(2048)<sup>n</sup>
+
+Output: the value-vector (_f<sub>1</sub>_(_x_), ... , _f<sub>n</sub>_(_x_))
+
+![$f_l(x) = \sum_{i=1}^{K} y_i[l] \prod_{\underset{j \neq i}{j=1}}^{K} \frac{x - x_j}{x_i - x_j}$](../assets/eip-3450/lagrange.gif)
+
+### Split Mnemonic
+
+Inputs: BIP-39 mnemonic, number of shares (_N_), threshold (_T_)
+
+Output: N Shares, each share including an id (_x_ ⊆ GF(2048)) and a BIP-39 mnemonic
+
+1. Check the following conditions:
+   - 1 < T <= N < 2048
+   - Each word in the mnemonic belongs to the word list
+     - Note: It isn't explicitly required that the mnemonic comply with the BIP-39 specification
+2. Convert the mnemonic to a value-vector, where each value is the index of the corresponding word in the word list
+3. Define values:
+   - Let _E_ be the value-vector representation of the mnemonic
+   - Let _n_ be the length of _E_
+   - Let _coeff<sub>1</sub>_, ... , _coeff<sub>T - 1</sub>_ ∈ GF(2048)_<sup>n</sup>_ be value-vectors generated randomly, independently with uniform distribution from a source suitable for generating cryptographic keys
+4. Evaluate the polynomial for each share
+   - For each _x_ from 1 to _N_, evaluate the polynomial _f(x)_ = _E_ + _coeff<sub>1</sub>x<sup>1</sup>_ + ... + coeff<sub>T - 1</sub>x<sup>T - 1</sup>, where _x_ is the share id and _f(x)_ is the share value (as a vector of integers)
+5. Define values:
+   - Let _x_ be the the share id
+   - Let _f(x)_ be _y_, the share value-vector
+6. Convert each share value-vector to a mnemonic using each value to lookup a word by index in the word list
+7. Return the id and mnemonic of each share
+
+### Recover Mnemonic
+
+Input: _K_ Shares, where _T_ <= K <= _N_
+
+Output: The original mnemonic
+
+1. Convert each share mnemonic to a value-vector, where each value is the index of the corresponding word in the word list
+2. Define values:
+   - Let _y<sub>k</sub>_ be the value-vector of share _k_
+   - Let _x<sub>k</sub>_ be the id of share _k_
+3. Calculate _E_ = Interpolate(0, [(_x<sub>1</sub>_, _y<sub>1</sub>_),...,(_x<sub>K</sub>_, _y<sub>K</sub>_)])
+   - See [Polynomial Interpolation](#polynomial-interpolation) above
+4. Convert _E_ to a mnemonic using each value to lookup a word by index in the word list
+
+## Test Cases
```

Comment:
```suggestion
## Rationale
TBD

## Test Cases
```
A rationale section is required.  It should indicate why specific decisions were made such as why a certain constant was picked over another, or why one algorithm was chosen over others.

Reply:
Added.